sd/lb/retry_test.go

```diff
@@ -88,3 +88,44 @@ func TestRetryTimeout(t *testing.T) {
 		t.Errorf("wanted %v, got none", context.DeadlineExceeded)
 	}
 }
+
+func AbortEarlyCustomMessage_WCB(t *testing.T) {
+    var (
+        cb = func(count int, msg string) (bool, *string) { 
+            ret := "Aborting early"
+            return false, &ret
+        }
+        endpoints = sd.FixedSubscriber{} // no endpoints
+        lb        = loadbalancer.NewRoundRobin(endpoints)
+        retry    = loadbalancer.RetryWithCallback(999, time.Second, lb, cb) // lots of retries
+        ctx       = context.Background()
+    )
+     _, err := retry(ctx, struct{}{})
+     if err == nil {
+        t.Errorf("expected error, got none") // should fail
+    }
+    if err.Error() != "Aborting early" {
```

Comment:
Comparing strings is not a great way to compare errors. You'll want the callback to accept and return a proper `error` type.